python/pyspark/broadcast.py

```diff
@@ -139,6 +140,24 @@ def __reduce__(self):
         return _from_id, (self._jbroadcast.id(),)
 
 
+class BroadcastPickleRegistry(threading.local):
```

Comment:
Using the lock was a little more obvious what is going on, but it's better to not use a lock in case of pickling a large command like @viirya said.  Also, this way doesn't need to change any of the pickling code, so I prefer it too.